lib/codecs/src/encoding/format/csv.rs

```diff
@@ -171,14 +373,148 @@ mod tests {
             ConfigTargetPath::try_from("field3".to_string()).unwrap(),
             ConfigTargetPath::try_from("field2".to_string()).unwrap(),
         ];
-        let config = CsvSerializerConfig::new(CsvSerializerOptions { fields });
+        let opts = CsvSerializerOptions {
+            fields,
+            ..Default::default()
+        };
+        let config = CsvSerializerConfig::new(opts);
         let mut serializer = config.build().unwrap();
         let mut bytes = BytesMut::new();
+
         serializer.encode(event, &mut bytes).unwrap();
 
         assert_eq!(
             bytes.freeze(),
             b"value1,value5,value5,value3,value2".as_slice()
         );
     }
+
+    #[test]
+    fn correct_quoting() {
+        let event = Event::Log(LogEvent::from(btreemap! {
```

Comment:
Also, let's encode at least two events to this and the following tests. It would be best if the events contain both numeric and non-numeric values. For convenience, consider creating a test utility to build events for your tests.

Reply:
💬 This is still relevant. Note that `encode` accepts `&mut self` so consecutive calls to it are potentially not idempotent.  Let's add another test case where we have multiple events and call encode on all of them.

Reply:
Nice, we now have `multiple_events()`. Resolving this thread.